core/src/main/scala/kafka/server/DynamicBrokerConfig.scala

```diff
@@ -675,7 +677,7 @@ object DynamicLogConfig {
   // Exclude message.format.version for now since we need to check that the version
   // is supported on all brokers in the cluster.
   @nowarn("cat=deprecation")
-  val ExcludedConfigs = Set(ServerLogConfigs.LOG_MESSAGE_FORMAT_VERSION_CONFIG)
```

Comment:
Do we need this variable? Maybe we can remove it by following change.
```scala
  // Exclude message.format.version for now since we need to check that the version
  // is supported on all brokers in the cluster.
  @nowarn("cat=deprecation")
  val ReconfigurableConfigs = ServerTopicConfigSynonyms.TOPIC_CONFIG_SYNONYMS.values.asScala.toSet - ServerLogConfigs.LOG_MESSAGE_FORMAT_VERSION_CONFIG
```

Reply:
I've inlined this variable but we need to keep it as a `Set` for the `-` operator to work

Reply:
I don't run it actually, but IIRC "-" operator should work. See following screenshot

![image](https://github.com/apache/kafka/assets/6234750/b2e46f3a-65b1-4e95-9aaf-ff48bbff11ad)

Please correct me for my Scala education 😄

Reply:
My bad, you are correct! I was trying to use the `--` operator, hence why it needed both operands to be `Set`s. With `-` we don't need the `Set`. Thanks